app/src/main/res/values/strings.xml

```diff
@@ -354,12 +354,14 @@
     <string name="layout_ics_darker"  translatable="false">Ice Cream Sandwich (dark hints)</string>
     <string name="layout_trans"  translatable="false">Transparent</string>
     <string name="layout_trans_neon"  translatable="false">Transparent (experimental)</string>
+	<string name="layout_material_black" translatable="false">Material Black</string>
     <string name="layout_material_dark" translatable="false">Material Dark</string>
     <string name="layout_material_light" translatable="false">Material Light</string>
 
     <string-array name="keyboard_layout_modes" translatable="false">
         <item>@string/layout_ics</item>
         <item>@string/layout_ics_darker</item>
+		<item>@string/layout_material_black</item>
```

Comment:
You also need to add a matching entry in keyboard_layout_modes_values below at the matching array position, using a new index (7). The actual preference is stored as an integer index into this array, and the indirection is needed to avoid misinterpreting existing themes.

Reply:
Correction, the stored preference is an integer index into the THEMES array in KeyboardSwitcher.java:

https://github.com/klausw/hackerskeyboard/blob/0e1a65fe37b70dcb0f6b4e0e49cd8a9a23011d46/app/src/main/java/org/pocketworkstation/pckeyboard/KeyboardSwitcher.java#L64

The new entry needs to be at the end in that array. keyboard_layout_modes_values needs to refer to that index, and the entries in that array and in keyboard_layout_modes need to match up pairwise.

(Sorry, this needs a clearer comment.)